File: org.summer.view.widget/src/org/summer/view/widget/controls/ContextMenuEventArgs.java
package org.summer.view.widget.controls;

import org.summer.view.widget.Delegate;
import org.summer.view.widget.DependencyObject;
import org.summer.view.widget.RoutedEventArgs;

/// <summary>
/// The data sent on a ContextMenuEvent 
/// </summary>
public /*sealed*/ class ContextMenuEventArgs extends RoutedEventArgs 
{ 
    /*internal*/ public ContextMenuEventArgs(Object source, boolean opening)
    {
        this(source, opening, -1.0, -1.0) ;
    }

    /*internal*/ public ContextMenuEventArgs(Object source, boolean opening, double left, double top) 
    {
        _left = left; 
        _top = top; 
        RoutedEvent =(opening ? ContextMenuService.ContextMenuOpeningEvent : ContextMenuService.ContextMenuClosingEvent);
        Source = source; 
    }

    /// <summary>
    ///     Position (horizontal) that context menu should displayed 
    /// </summary>
    public double CursorLeft 
    { 
        get { return _left; }
    } 

    /// <summary>
    /// Position (vertical) that context menu should displayed
    /// </summary> 
    public double CursorTop
    { 
        get { return _top; } 
    }

    /*internal*/ public DependencyObject TargetElement
    {
        get { return _targetElement; }
        set { _targetElement = value; } 
    }

    /// <summary> 
    /// Support DynamicInvoke for ContextMenuEvent
    /// </summary> 
    /// <param name="genericHandler"></param>
    /// <param name="genericTarget"></param>
    protected /*override*/ void InvokeEventHandler(Delegate genericHandler, Object genericTarget)
    { 
        ContextMenuEventHandler handler = (ContextMenuEventHandler)genericHandler;
        handler(genericTarget, this); 
    } 

    private double _left; 
    private double _top;
    private DependencyObject _targetElement;
}

File: summer/WebContent/js/system/windows/controls/ContextMenuEventArgs.js
/**
 * ContextMenuEventArgs
 */

define(["dojo/_base/declare", "system/Type", "windows/RoutedEventArgs"], function(declare, Type){
	var ContextMenuEventArgs = declare("ContextMenuEventArgs", RoutedEventArgs,{
		constructor:function(/*object*/ source, /*bool*/ opening, /*double*/ left, /*double*/ top) 
        {
			if(left === undefined){
				left = -1.0;
			}
			
			if(top === undefined){
				top = -1.0;
			}
			
			this._targetElement = null;
            this._left = left; 
            this._top = top; 
            this.RoutedEvent =(opening ? ContextMenuService.ContextMenuOpeningEvent : ContextMenuService.ContextMenuClosingEvent);
            this.Source = source; 
        },
        
        /// <summary> 
        /// Support DynamicInvoke for ContextMenuEvent
        /// </summary> 
        /// <param name="genericHandler"></param>
        /// <param name="genericTarget"></param>
//        protected override void 
        InvokeEventHandler:function(/*Delegate*/ genericHandler, /*object*/ genericTarget)
        { 
            /*ContextMenuEventHandler*/var handler = /*(ContextMenuEventHandler)*/genericHandler;
            handler.Invoke(genericTarget, this); 
        } 
	});
	
	Object.defineProperties(ContextMenuEventArgs.prototype,{
	       /// <summary>
        ///     Position (horizontal) that context menu should displayed 
        /// </summary>
//        public double 
        CursorLeft: 
        { 
            get:function() { return this._left; }
        },

        /// <summary>
        /// Position (vertical) that context menu should displayed
        /// </summary> 
//        public double 
        CursorTop:
        { 
            get:function() { return this._top; } 
        },
 
//        internal DependencyObject 
        TargetElement:
        {
            get:function() { return this._targetElement; },
            set:function(value) { this._targetElement = value; } 
        }
	});
	
	ContextMenuEventArgs.Type = new Type("ContextMenuEventArgs", ContextMenuEventArgs, [RoutedEventArgs.Type]);
	return ContextMenuEventArgs;
});
